pkg/discovery-manager/discovery_controller.go

```diff
@@ -47,13 +56,309 @@ type DiscoveryReconciler struct {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	case nodecorev1alpha1.PhaseFailed:
+		klog.Infof("Discovery %s failed, no Peering Candidate found", discovery.Name)
+		setDiscoveryPhase(&discovery, nodecorev1alpha1.PhaseFailed)
+		if err := r.updateSolverStatus(ctx, &discovery); err != nil {
+			klog.Errorf("Error when updating Solver %s status: %s", req.NamespacedName, err)
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
 
 	return ctrl.Result{}, nil
 }
 
+// setDiscoveryPhase sets the phase of the discovery
+func setDiscoveryPhase(d *advertisementv1alpha1.Discovery, phase nodecorev1alpha1.Phase) {
+	d.Status.Phase.Phase = phase
+	d.Status.Phase.LastChangeTime = time.Now().Format(time.RFC3339)
+}
+
+// updateSolverStatus updates the status of the discovery
+func (r *DiscoveryReconciler) updateSolverStatus(ctx context.Context, discovery *advertisementv1alpha1.Discovery) error {
+	return r.Status().Update(ctx, discovery)
+}
+
+// getFlavour is a function that returns an array of Flavour that fit the Selector by performing a get request to an http server
+func getFlavour(selector nodecorev1alpha1.FlavourSelector) ([]*nodecorev1alpha1.Flavour, error) {
+	// Marshal the selector into JSON bytes
+	cpuInt, _ := selector.Cpu.AsInt64()
+	ramInt, _ := selector.Memory.AsInt64()
+	ephStorage, _ := selector.EphemeralStorage.AsInt64()
+
+	body := map[string]interface{}{
+		"cpu":    int(cpuInt),
+		"ram":    int(ramInt),
+		"memory": int(ephStorage),
+		"type":   K8S_TYPE,
+	}
+
+	klog.Info("Selector is ", body)
+	selectorBytes, err := json.Marshal(body)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create the Flavour CR from the first flavour in the array of Flavour
+	var flavoursCR []*nodecorev1alpha1.Flavour
+
+	// Send the POST request to all the servers in the list
+	for _, ADDRESS := range SERVER_ADDRESSES {
+		resp, err := http.Post(ADDRESS+"/listflavours/selector", "application/json", bytes.NewBuffer(selectorBytes))
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+
+		// Check if the response status code is 200 (OK)
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("received non-OK response status code: %d", resp.StatusCode)
+		}
+
+		// Read the response body
+		respBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		// Unmarshal the response JSON into an array of Flavour
+		var flavours []*nodecorev1alpha1.Flavour
+		if err := json.Unmarshal(respBody, &flavours); err != nil {
+			return nil, err
+		}
+
+		for _, flavour := range flavours {
+			klog.Infof("Flavour found: %s", flavour.Name)
+			cr := createFlavourCustomResource(flavour)
+			flavoursCR = append(flavoursCR, cr)
+		}
+
+	}
+	klog.Info("Flavours created", flavoursCR)
+	return flavoursCR, nil
+
+}
+
+// createFlavourCustomResource creates a Flavour CR from a Flavour
+func createFlavourCustomResource(flavour *nodecorev1alpha1.Flavour) *nodecorev1alpha1.Flavour {
```

Comment:
`forgeFlavour`. Moreover I have already seen this function somewhere (maybe localResourceManager). We have not to do duplicate code!

Reply:
Two different functions, one create a Flavour from a NodeInfo struct...

Reply:
and the name of the function?

Reply:
Fixed.